atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/TimestampChecker.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.jepsen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import clojure.lang.PersistentArrayMap;
+import clojure.lang.PersistentVector;
+
+public final class TimestampChecker {
+    private static final Logger log = LoggerFactory.getLogger(TimestampChecker.class);
+
+    private TimestampChecker() {
+    }
+
+    public static boolean checkClojureHistory(PersistentVector clojureHistory) {
+        History history = parse(clojureHistory);
+        if (history == null) {
+            return false;
+        }
+
+        return checkHistory(history);
+    }
+
+    private static History parse(PersistentVector clojureHistory) {
+        try {
+            List<ParsedMap> parsed = parseIntoJavaObjects(clojureHistory);
+            return parseIntoEvents(parsed);
+        } catch (Exception e) {
+            log.error("Failed to parse output from Jepsen : " + e);
+            return null;
+        }
+    }
+
+    private static boolean checkHistory(History history) {
+        try {
+            MonotonicChecker monotonicChecker = new MonotonicChecker();
```

Comment:
I'm thinking in the long run we have a `Set<Visitor> checkers` and then do `checkers.forEach(checker -> history.accept(checker))`, failing if any of them fails?